src/services/twilio-api/get-help-doc-link.js

```diff
@@ -0,0 +1,60 @@
+const jsonMap = require('../api-doc-mapping.json');
+// url: {path : {topics: {}, commands: {}}, subpath : {wherever applicable}}
+const urlMap = new Map(Object.entries(jsonMap));
+
+const getSubroute = (cmdDetailsArr, subpathMap) => {
+  const subdomain = cmdDetailsArr[3];
+  // apiName/version to locate sub url in json map
+  const apiVersion = `${cmdDetailsArr[1]}/${cmdDetailsArr[2]}`;
+  const subUrlArr = subpathMap.get(apiVersion);
+  if (subUrlArr === undefined) {
+    return '';
+  }
+  const subUrlMap = new Map(Object.entries(subUrlArr));
+  const subUrl = subUrlMap.get(subdomain);
+  return subUrl;
+};
+
+const getRootPath = (cmdDetailsArr, subpathMap, topicsMap, commandsMap) => {
+  const topicName = cmdDetailsArr[0];
+  const apiName = cmdDetailsArr[1];
+  let url = '';
+
+  if (topicName === 'api') {
+    // get the array of apis and convert to a map
+    const arrOfAllApis = topicsMap.get(topicName);
+    if (arrOfAllApis === null) {
+      return '';
+    }
+    const mapOfAPIs = new Map(Object.entries(arrOfAllApis));
+    url = mapOfAPIs.get(apiName);
+    url += getSubroute(cmdDetailsArr, subpathMap) || '';
+  } else {
+    // its a command or another topic
+    url = topicsMap.get(topicName) || commandsMap.get(topicName);
+  }
+
+  return url || '';
+};
+
+const getDocLink = (key) => {
+  const pathMap = new Map(Object.entries(urlMap.get('path')));
+  const subpathMap = new Map(Object.entries(urlMap.get('subpath')));
+  const topicsMap = new Map(Object.entries(pathMap.get('topics')));
+  const commandsMap = new Map(Object.entries(pathMap.get('commands')));
+
+  const baseUrl = 'https://twilio.com/docs/';
+  let url = '';
+  const cmdDetailsArr = key.split(':'); // original command array
+  const cmd = cmdDetailsArr[0];
+  if (topicsMap.has(cmd) || commandsMap.has(cmd)) {
+    url = getRootPath(cmdDetailsArr, subpathMap, topicsMap, commandsMap);
+  } else {
+    url = 'rest/';
+  }
+  return baseUrl + (url || '');
+};
+
+module.exports = {
+  getDocLink,
+};
```

Comment:
Are the unit tests still pending for this?

Reply:
Added [here](https://github.com/twilio/twilio-cli/blob/c42df8ba8850be5f826d6f8950fe164af19a3089/test/services/twilio-api.test.js#L159).